crates/solver/src/liquidity/zeroex.rs

```diff
@@ -138,13 +175,18 @@ fn get_useful_orders(order_buckets: OrderBuckets, orders_per_type: usize) -> Vec
 struct OrderSettlementHandler {
     order: Order,
     zeroex: IZeroEx,
+    allowances: Arc<Allowances>,
 }
 
 impl SettlementHandling<LimitOrder> for OrderSettlementHandler {
     fn encode(&self, executed_amount: U256, encoder: &mut SettlementEncoder) -> Result<()> {
         if executed_amount > u128::MAX.into() {
             anyhow::bail!("0x only supports executed amounts of size u128");
         }
+        encoder.append_to_execution_plan(
```

Comment:
What would happen if we approved already zeroEx.address() with 5 tokens. And we have 2 different orders each selling 4 tokens? Then I assume we would get always Allowance::Sufficient and not set an approval, but the tx would fail as the approval is not sufficient.

I assume that it's not really an issue, as we assume that we have an infinitive approval set, right?

Reply:
AFAICS you are correct. I think all the liquidity implementations currently have this problem but it can be fixed for all by a patch in the `Allowances` implementation. Every time we get the approval for a token we could immediately deduct the approved amount and if a later interaction would put the allowance over the edge an approval interaction would be issued.